dataGeneration/generate-cosine-data-multi-entity.py

```diff
@@ -0,0 +1,296 @@
+                nextTs = startTs
+                prb = Random()
+                prb.seed(random.randint(0, 100000000))
+                cosine_p = cosine_params[index]
+                data_index = 0
+                for i in range(0, time_intervals):
+                    ts = nextTs.strftime(dtFormat)
+                    for k in range(0, sample_per_interval):
+                        data = generate_val(cosine_p[1], cosine_p[0], 2, data_index,
+                                            50, 5, prb)
+                        bulk_payload.append(
+                            {
+                                index_name: index_value,
+                                "_source":
+                                    {
+                                        timestamp_name: ts,
+                                        cpu_name: data[0],
+                                        mem_name: data[1],
+                                        host_name: host_str,
+                                        process_name: process_str
+                                    }
+                            }
+                        )
+                        count += 1
+                        data_index += 1
+                        if count >= batch_size:
+                            post_log(bulk_payload, thread_index)
+                            bulk_payload = list()  # reset list
+                            totalCount += count
+                            count = 0
+                    # increment by ingestion_frequency (in seconds) after looping through each host multiple samples
+                    nextTs += datetime.timedelta(seconds=INGESTION_FREQUENCY)
+                    if totalCount - lastTotalCount > 1000000:
+                        # report progress every 1 million inserts
+                        print("totalCount {} thread_index {}".format(totalCount,
+                                                                     thread_index))
+                        lastTotalCount = totalCount
+            j += 1
+
+        if len(bulk_payload) > 0:
+            post_log(bulk_payload, thread_index)
+    except Error as err:
+        print("error: {0}".format(err))
+
+def split(a, n):
+    k, m = divmod(len(a), n)
+    return (a[i*k+min(i, m):(i+1)*k+min(i+1, m)] for i in range(n))
+
+# create an list of array of size total_entities, the inner array has 2 subarrays: phase, amp
+def create_cosine(total_entities, base_dimension, period, amplitude):
+    cosine_param = np.empty(total_entities, dtype=object)
+    for i in range(0, total_entities):
+        phase = np.empty(base_dimension, dtype=float)
+        amp = np.empty(base_dimension, dtype=float)
+        for j in range(0, base_dimension):
+            phase[j] = random.randint(0, period)
+            amp[j] = (1 + 0.2 * random.random()) * amplitude
+        cosine_param[i] = np.array([phase, amp])
+    return cosine_param
+
+'''
+    Main entry method for script
+'''
+def main():
```

Comment:
This part might needs refactoring as we need to extract a method create_client to be used elsewhere. See above comments.

Reply:
will make the changes after trying out this changes on enough points, will run it over the weekend.